File: src/main/java/com/paqi/friendsystem/service/DiscussService.java
package com.paqi.friendsystem.service;

import com.paqi.friendsystem.entity.Discuss;

import java.util.ArrayList;

/**
 * @author PQ
 * @Description 评论服务接口
 * @Date 下午8:34 17/3/2020
 * @version 3.3.0
**/
public interface DiscussService {
    /**
     * 添加一条新评论
     * @author PQ
     * @Description 添加一条新评论
     * @param discuss：评论实体类
     * @return 返回添加成功的评论id
     * @Date 下午8:35 17/3/2020
     * @version 3.3.0
    **/
    int addDiscuss(Discuss discuss);

    /**
     * 根据文章id获取评论列表
     * @author PQ
     * @Description 根据文章id获取评论
     * @param literatureId：文章id
     * @return 返回评论列表
     * @Date 17:17 22/3/2020
     * @version
    **/
    ArrayList<Discuss> getDiscussList(int literatureId);
}

File: src/main/resources/static/js/service/discussService.js
/**
 * @author PQ
 * @Description 评论服务类
 * @Date 16:53 22/3/2020
 * @version 3.4.24
 **/

/**
 * @author PQ
 * @Description 写一个评论
 * @param discuss:示例
 *      {
            "literatureId": literatureId,
            "context": "context"
        }
 * @param func：func(boolean)
 * @Date 16:54 22/3/2020
 * @version 3.4.24
 **/
function writeDiscuss(discuss, func) {
    $.ajax({
        url: "/api/literature/write-discuss",
        type: "POST",
        data: discuss,
        success: function (boolean) {
            func(boolean);
        },
        error: function (msg) {
            console.log("发送失败");
        }
    });
}

/**
 * @author PQ
 * @Description 根据文章id获取文章list
 * @param：literatureId: {
 *     "literatureId": literatureId
 * }
 * @param:func: func(list)
 * @Date 17:23 22/3/2020
 * @version
**/
function getDiscussList(literatureId, func) {
    $.ajax({
        url: "/api/literature/write-discuss",
        type: "GET",
        data: literatureId,
        success: function (list) {
            func(list);
        },
        error: function (msg) {
            console.log("发送失败");
        }
    });
}